src/checks/series/monotonic_decrease_check.rs

```diff
@@ -0,0 +1,227 @@
+use crate::{DataCache, Error, Flag, Timeseries};
+
+/// Timeseries check that detects if a timeseries is monotonically slightly decreasing in a given
+/// window. Useful for detecting evaporation/leakage of a bucket measuring precipitation.
+///
+/// Takes a series of data points representing the window to check for increases, and upper and
+/// lower limits that the total difference over the whole series must be within to meet the
+/// failure condition
+///
+/// Since this check is really looking at the intervals between data points, If you are passing in
+/// regularly spaced chunks of a time series (e.g. each day's hourly values), you should include
+/// an extra point of overlap, so that you check all relevent intervals (e.g. in the aformentioned
+/// example you should pass in 25 points, including both the 00:00 and 24:00 points)
+///
+/// Returns:
+/// - [`Flag::DataMissing`] if any of the data points are missing,
+/// - [`Flag::Fail`] if there are no increases between consecutive data points in the series
+///   AND the total decrease over the series is within the given range
+/// - [`Flag::Pass`] otherwise.
+pub fn monotonic_decrease_check(data: &[Option<f32>], lower_limit: f32, upper_limit: f32) -> Flag {
+    if data.iter().any(Option::is_none) {
+        return Flag::DataMissing;
+    }
+
+    // take a rolling windows of 2 datapoints over the dataset
+    // if the later value is less than the earlier for any of them, there was a decrease
+    // apply `!` since we want `true` if there is no decrease
+    let no_increase = !data.windows(2).any(|window| window[1] > window[0]);
+
+    let total_diff = data[0].unwrap() - data[24].unwrap();
+
+    if (no_increase) && (lower_limit..=upper_limit).contains(&total_diff) {
+        Flag::Fail
+    } else {
+        Flag::Pass
+    }
+}
+
+/// Apply [`monotonic_decrease_check`] to a whole [`DataCache`]
+///
+/// `window_size` here refers to the number of intervals to consider at once, as this is more
+/// conceptually straightforward than the number of points (see docs for
+/// [`monotonic_decrease_check`]). Each iteration of `monotonic_decrease_check` will actually have
+/// `window_size` + 1 points passed to it.
+///
+/// As `window_size` predecessors to each observation are needed, the [`SeriesCache`]
+/// provided must have `num_leading_points` >= `window_size`
+///
+/// ## Errors
+///
+/// - data is invalid
+/// - data has `num_leading_points` < `window_size`
```

Comment:
FWIW this is not actually how it's done in kvalobs today. I believe they run this check daily on 00:00-24:00 of hourly data. We could simulate that by using chunks instead of windows, and then adding an extra point of context to each chunk, and using std::iter::repeat_n to expand the flag to cover all points in the chunk.

The version I wrote here is more computationally intensive, but I wrote it like this because I think it's more consistent in the way it flags. In the chunk version, periods of monotonic decreases only need be as long as `window_size` if they perfectly align with the chunk boundaries, but they may need to be as long as `2 * window_size - 1` if they are badly aligned.

Reply:
There is perhaps a case to be made for expanding the Fail flags backwards by `window_size` to cover the whole region that failed the check... 🤔

Reply:
Unless I completely misunderstand the whole system, we don't have to modify the algo to simulate kvalobs, you could simply pass in a `DataCache` that contains "daily" timeseries? Anyway I agree, this check is more consistent than the one in kvalobs.

Reply:
> There is perhaps a case to be made for expanding the Fail flags backwards by `window_size` to cover the whole region that failed the check... 🤔

One flag per data point instead of per window?

Reply:
Mmm, no, it's one per window (look at the test?). But that's the same for all other checks that scan by window :sweat_smile:

Reply:
Yeah, so just apply the window flag result to all the points in the each window and figure out some flag precedence rule for the points that are part of multiple windows?

Reply:
Looks right!

Reply:
Ok, this is done, and I also revamped the test and doc comment. Let me know if it makes more sense now.

This change raised a couple of new concerns though:
1. `flatline_check` should probably get the same treatment
2. This change means the check now requires `window_size` trailing points as well as leading ones, which worsens the restriction on how long we have to wait to QC a piece of data. Should we offer an option that doesn't require trailing points and instead pads with `None`s? I'll bring it up in the WP5 meeting tomorrow.

Reply:
I've made issue #56 to track the conclusion of our discussion today. I think we should leave it for a later PR though